allow fetching of specific versions

Tasketo opened the pull request:
This allows to fetch a specific version of a package like `node cli.js <package>@<version>`.

Usecase: Legacy projects with not up2date dependencies, where licenses might be changed in the meanwhile.

Tasketo commented:
@franciscop please review and merge :)

franciscop commented:
This doesn't seem like it'd work with namespaced packages like [`@sindresorhus/slugify`](https://github.com/sindresorhus/slugify).

Tasketo commented:
Good catch! I fixed that and tested with:
```
node cli.js legally 
node cli.js legally@3.3.3
node cli.js @sindresorhus/slugify
node cli.js @sindresorhus/slugify@0.1.0
```
I also adjusted the temp-directory structure to prevent another level (cause of the `/`)
The example above results in:
```
$ ls
pack-_sindresorhus_slugify/        pack-legally/
pack-_sindresorhus_slugify_0.1.0/  pack-legally_3.3.3/
```

franciscop commented:
I'm not sure about replacing `@` and `/` with `_`, this might lead to package collisions (e.g. `hello@1` is the same as `hello_1`). I don't know how rare these might be, but why take the risk? Having nested folders on the other hand should avoid collisions, what happens if you don't include the last `.replace()`? Does it still work?

Tasketo commented:
I don't know a single one, but let's make it more unlikely -> again with multi level

The replacement of `@` is required, when a version is given. Else npm will fail with `npm ERR! Invalid name: "slugify@0.1.0"\`

franciscop commented:
Awesome, thanks. I am having some trouble with the build right now so I will need to publish it next week.

Tasketo commented:
> Awesome, thanks. I am having some trouble with the build right now so I will need to publish it next week.

Thanks for merging and friendly reminder for npm publishing @franciscop

franciscop commented:
Thanks! Published as `legally@3.5.9`! :tada: 

I also created a lightweight version for quick license checks, the dependency tree should be more accurate but you need to have a `package-lock.json` to use it:
- https://www.npmjs.com/package/check-licenses
